### ISSUE TITLE
Active Response Not Triggering 

### ISSUE THREAD

**Issue by wehde92** [wazuh/wazuh#11203]
| Wazuh version 4.2.5-1 | Active Response | Ubuntu Package | Ubuntu 20.04 |

I have wazuh running inside a ubuntu 20.04 LXC container. So far everything works except for the alert trigger that normally would call the active response.

Manually calling agent-control works

![image](https://user-images.githubusercontent.com/8854624/144673971-50ac5286-f788-4d59-8c9c-d67bbe8b78b5.png)

The email gets generated and sent on the correct event trigger:

![image](https://user-images.githubusercontent.com/8854624/144674338-7d0971b4-ceea-49d9-85e8-850944e8563e.png)

I suspect it might have something to do with attaching to the /var/ossec/queue/alerts/ sockets from within LXC.

The ossec.log with debug turned on doesn't show much for errors.

**Comment by wehde92** [comment]
Here is the etc/shared/group/agent.conf

![image](https://user-images.githubusercontent.com/8854624/144675048-e0537d3f-af7a-481b-a90d-eafb60cfefec.png)

**Comment by wehde92** [comment]
The only notable error with debug turned on:

![image](https://user-images.githubusercontent.com/8854624/144675822-984d436b-d2eb-404c-9e3d-bbb31b2a8070.png)

**Comment by wehde92** [comment]
I have made some more progress on this. It looks like you cannot specify a Command or Active-Response bock inside the etc/shared/group/agent.conf.

Here is the output of my rds groups agent.conf:

<img width="565" alt="image" src="https://user-images.githubusercontent.com/8854624/144911387-cbe59883-64fb-433b-b7de-446a34e5e0b7.png">

Then here is the result from agent_control:

<img width="402" alt="image" src="https://user-images.githubusercontent.com/8854624/144911456-6da8aaa6-ac36-4c75-8900-9d9473051c95.png">

Furthermore, when adding to the etc/ossec.conf file for the command and AR blocks, you will notice the rule names do not get built properly and passed to the client properly.

<img width="337" alt="image" src="https://user-images.githubusercontent.com/8854624/144911814-567b68bf-c340-4738-920f-f66003fa99ce.png">

My command becomes win_route-null2900

<img width="481" alt="image" src="https://user-images.githubusercontent.com/8854624/144911884-ba14856b-85d8-4c5d-8854-8ccdde1a47ef.png">

I'm guessing when it builds the command it is combining the Name field with the timeout field.

**Comment by vikman90** [comment]
Hi @wehde92,

You're right, the `<active-response>` configuration must be declared in the manager's _ossec.conf_.

This is the configuration that we need to add to the **agent** in order to enable Active Response:
```xml
  <active-response>
    <disabled>no</disabled>
  </active-response>
```

The internal name of the AR command is indeed the name and the timeout:
```sh
cat /var/ossec/etc/shared/ar.conf

restart-ossec0 - restart-ossec.sh - 0
restart-ossec0 - restart-ossec.cmd - 0
restart-wazuh0 - restart-ossec.sh - 0
restart-wazuh0 - restart-ossec.cmd - 0
restart-wazuh0 - restart-wazuh - 0
restart-wazuh0 - restart-wazuh.exe - 0
demo1 - demo - 1
```

This file maps AR commands to executable files and timeouts. This allows the agent to run multiple versions of the same AR command with different timeouts. I checked that adding a trailing number to the `<name>` option works.

Regarding the error, that is a known issue that affects containerized environments and Valgrind, but that should not be a problem except when running a highly loaded manager.

In summary:
- Define `<command>` and `<active-response>` commands in the manager.
- Just enable `<active-response>` in the agents.

If this does not work, please share with us your current agent and manager configuration, and a sample alert for us to reproduce the issue.

Best regards.

**Comment by wehde92** [comment]
Thanks @vikman90 for your quick response.

Is there a way to reload the manager ossec.conf after updating rules without triggering a full wazuh-control restart?

I was able to get AR running now by building a full set of <command> and <active-response> blocks in the main ossec.conf. 

It might be nice to be update the docs to notate what configs are monitored in the etc/shared/ directory for future reference.

**Comment by vikman90** [comment]
@wehde92 we recently improved the manager to support ruleset reloading, but this works for log testing only. Currently, we need to fully restart the manager to apply the configuration. On the other hand, we plan to support dynamic ruleset reload (#39).

Best regards.
